xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
```

Comment:
I've used variable name `s` for parameter s in line 90:

`double s = unserialise_double(&ptr, end);`

Should I use some other name for it? May be using original name i.e. `slope` would be more appropriate.